template.js

```diff
@@ -1,5 +1,37 @@
-let button = document.querySelector('button');
+let myHeaders2 = new Headers();
+myHeaders2.append("Origin", "https://localhost.com");
 
-button.onclick = function () {
-  open("index.html", "_self");
-};
\ No newline at end of file
+let requestOptions2 = {
+  method: 'GET',
+  headers: myHeaders2,
+  redirect: 'follow'
+};
+let arrayPromise2 = fetch("https://image-gallery-pjks.s3.ca-central-1.amazonaws.com/data.json", requestOptions2)
+  .then(res => res.json())
+  .then(res => {
+    console.log(res)
+    return res;
+  });
```

Comment:
you are setting the `id` AND `array` in the sessionStorage, but then you are making a whole API call to get the array again.

Instead, how about either just searching the array that you have stored in the sessionStorage OR just having the  `url`, `descreption`, `location` within sessionStorage.